lib/phoenix/router/live_reloader.ex

```diff
@@ -36,12 +36,14 @@ defmodule Phoenix.Router.LiveReload do
   defp html_content_type?([]), do: false
   defp html_content_type?([type | _]), do: String.starts_with?(type, "text/html")
 
-  defp reload_assets_tag() do
+  defp reload_assets_tag(conn) do
+    config = conn.private.phoenix_endpoint.config(:live_reload)
+    url = config[:url] || ""
     """
     <script>
       #{@phoenix_js}
       var phx = require("phoenix")
-      var socket = new phx.Socket("/phoenix")
+      var socket = new phx.Socket("#{url}/phoenix")
```

Comment:
Path.join should be used here, else trailing/non-trailing url's will be problematic